WordPress/Sniffs/PHP/DisallowAlternativePHPTagsSniff.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     https://make.wordpress.org/core/handbook/best-practices/coding-standards/
+ */
+
+/**
+ * Verifies that no alternative PHP open tags are used.
+ *
+ * If alternative PHP open tags are found, this sniff can fix both the open and close tags.
+ *
+ * @link     https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/580
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @author   Juliette Reinders Folmer <wpplugins_nospam@adviesenzo.nl>
+ */
+class WordPress_Sniffs_PHP_DisallowAlternativePHPTagsSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		$tokens = array(
+			T_OPEN_TAG,
+			T_OPEN_TAG_WITH_ECHO,
+		);
+
+		$asp_enabled   = (boolean) ini_get( 'asp_tags' );
+		$short_enabled = (boolean) ini_get( 'short_open_tag' );
+
+		if ( false === $asp_enabled || ( true === $asp_enabled && false === $short_enabled ) ) {
+			$tokens[] = T_INLINE_HTML;
+		}
+
+		return $tokens;
+
+	} // end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens        = $phpcsFile->getTokens();
+		$openTag       = $tokens[ $stackPtr ];
+		$asp_enabled   = (boolean) ini_get( 'asp_tags' );
+		$short_enabled = (boolean) ini_get( 'short_open_tag' );
```

Comment:
Why do we base the behavior of the sniff on the ini settings of the PHP build that is running PHPCS? If you ask me that kind of defeats the purpose of the sniff, which is to check that you are using cross-site compatible PHP tags (at least that's what I *think* one purpose of it is). So wouldn't it be better to check for all tags by default, but allow this to be configured in custom rulesets if folks want to allow some of them? Or am I missing something here? At the very least the settings should be retrieved only once, for better performance.